placement/algo/sharded_helper.go

```diff
@@ -126,7 +131,19 @@ func newAddInstanceHelper(
 	}
 
 	return newHelper(p, p.ReplicaFactor(), opts), instanceInPlacement, nil
+}
 
+func newAddInstanceHelper(
```

Comment:
This looks quite similar to `newAddEmptyInstanceHelper`, what's the difference between the two? When would we use one or the other?

Reply:
newAddEmptyInstanceHelper only adds instance that is not in the placement or is leaving in the placement, basically has no load on it.